javascript/ql/src/semmle/javascript/security/dataflow/CleartextLogging.qll

```diff
@@ -42,6 +42,24 @@ module CleartextLogging {
         write.getRhs() = src and
         trg.(DataFlow::SourceNode).flowsTo(write.getBase())
       )
+      or
+      // Taint step through `util.inspect(..)` from Node.js
+      trg = DataFlow::moduleImport("util").getAMethodCall("inspect") and
+      trg.(DataFlow::CallNode).getAnArgument() = src
+      or
+      // Taint step through a `str.replace(..)` call.
+      trg.(DataFlow::MethodCallNode).getCalleeName() = "replace" and
+      trg.(DataFlow::MethodCallNode).getReceiver() = src
+      or
+      // Taint through the arguments object.
+      exists(DataFlow::CallNode call, Function f, VarAccess var |
+        src = call.getAnArgument() and
+        f = call.getACallee() and
+        not call.isImprecise() and
+        var.getName() = "arguments" and
+        var.getContainer() = f and
+        trg.asExpr() = var
```

Comment:
`trg = f.getArgumentsVariable().getAnAccess()` (though as mentioned during the call today a solution that directly reasons about forwarding calls would be even nicer, provided it's not too complex and/or expensive)

Reply:
I think I prefer this style of doing it.  
The initial commit had a [class for calls that forward to console.log](https://github.com/Semmle/ql/blob/1b90d5a44ec39197762eabfa6d1015b389d03a08/javascript/ql/src/semmle/javascript/frameworks/Logging.qll#L78).